File: Level1/신고_결과_받기/solution.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <set>

using namespace std;

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
    vector<int> answer;
    int id_cnt = id_list.size();
    unordered_map<string, set<string>> from_hash_table;
    unordered_map<string, set<string>> to_hash_table;
    for (auto id: id_list) {
        from_hash_table.insert(make_pair(id,  set<string>()));
        to_hash_table.insert(make_pair(id,  set<string>()));
    }
    
    for (auto line: report) {
        auto space = line.find(' ');
        auto id1 = line.substr(0, space);
        auto id2 = line.substr(space + 1, line.size());
        from_hash_table[id1].insert(id2);
        to_hash_table[id2].insert(id1);
    }
    
    for (auto id: id_list) {
        int cnt = 0;
        for (auto reported: from_hash_table[id]) {
            if (to_hash_table[reported].size() >= k)
                cnt++;
        }
        answer.push_back(cnt);
    }
    return answer;
}

File: Level1/신고_결과_받기/solution.java
import java.util.HashMap;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int id_list_len = id_list.length;
        int[] answer = new int[id_list_len];
        HashMap<String,Integer> ht = new HashMap<String,Integer>();
        boolean[][] flags = new boolean[id_list_len][id_list_len];
        int[] counts = new int[id_list_len];
        
        for (int i = 0; i < id_list_len; i++) {
            answer[i] = 0;
            ht.put(id_list[i], i);
            for (int j = 0; j < id_list_len; j++)
                flags[i][j] = false;
            counts[i] = 0;
        }
        for (String rep: report) {
            String[] r = rep.split(" ");
            int i = ht.get(r[0]);
            int j = ht.get(r[1]);
            if (!flags[j][i]) {
                flags[j][i] = true;
                counts[j]++;
            }
        }
        for (int i = 0; i < id_list_len; i++) {
            if (counts[i] >= k) {
                for (int j = 0; j < id_list_len; j++) {
                    if (flags[i][j])
                        answer[j]++;
                }
            }
        }
        return answer;
    }
}